### PR TITLE
refactor: taxonomy list

### PR DESCRIPTION
**Pull request by rohitp-bsf**
### Description
- refactor: taxonomy list

### Screenshots
- NA

### Types of changes
- Converted Class-based components to functional-based.
- Refactored the edit JS file into render and settings.

### How has this been tested?
- NA

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/taxonomy-list/settings.js [hunk 1]
```diff
@@ -0,0 +1,1419 @@
...
+			},
+		};
+
+		loadListGoogleFonts = (
+			<WebfontLoader config={ listconfig }></WebfontLoader>
+		);
+	}
+
+	// All Controls.
+	const color_control = (
```
Comment (RajkiranBagal) [inline src/blocks/taxonomy-list/settings.js:199]: camelCase variables.

File: src/blocks/taxonomy-list/settings.js [hunk 2]
```diff
@@ -0,0 +1,1419 @@
...
+						onChange={ ( colorValue ) =>
+							setAttributes( { listStyleColor: colorValue } )
+						}
+						allowReset
+					/>
+				</>
+			) }
+		</>
+	);
+	const color_control_hover = (
```
Comment (RajkiranBagal) [inline src/blocks/taxonomy-list/settings.js:243]: camelCase variables.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/taxonomy-list/block.js [hunk 3]
```diff
@@ -3,34 +3,34 @@
...
+import './style.scss';
+import './editor.scss';
+import edit from './edit';
 
 // Components
 import { __ } from '@wordpress/i18n';
 
 // Register block controls
-const { registerBlockType } = wp.blocks
+const { registerBlockType } = wp.blocks;
```
Comment (sandeshjangam) [inline src/blocks/taxonomy-list/block.js:17]: Use package

File: src/blocks/taxonomy-list/edit.js [hunk 4]
```diff
@@ -1,932 +1,103 @@
...
-	}
-	
-	onSelectTaxonomyType( value ) {
-		const { setAttributes } = this.props
-
-		setAttributes( { taxonomyType: value } )
-		setAttributes( { categories: "" } )
-	}
-	componentDidMount() {
+const { withSelect } = wp.data;
```
Comment (sandeshjangam) [inline src/blocks/taxonomy-list/edit.js:19]: Use package

File: src/blocks/taxonomy-list/settings.js [hunk 5]
```diff
@@ -0,0 +1,1419 @@
...
+const {
+	PanelBody,
+	RangeControl,
+	SelectControl,
+	TabPanel,
+	TextControl,
+	Button,
+	ButtonGroup,
+	ToggleControl,
+} = wp.components;
```
Comment (sandeshjangam) [inline src/blocks/taxonomy-list/settings.js:35]: Use package

File: src/blocks/taxonomy-list/settings.js [hunk 6]
```diff
@@ -0,0 +1,1419 @@
...
+
+const { InspectorControls, ColorPalette } = wp.blockEditor;
```
Comment (sandeshjangam) [inline src/blocks/taxonomy-list/settings.js:37]: Use package

**Review by sandeshjangam — CHANGES_REQUESTED** [review]
Add the package changes

File: src/blocks/taxonomy-list/settings.js (hunk 1 above)
Comment (rohitp-bsf) [reply src/blocks/taxonomy-list/settings.js:199]: Done!

File: src/blocks/taxonomy-list/settings.js (hunk 2 above)
Comment (rohitp-bsf) [reply src/blocks/taxonomy-list/settings.js:243]: Done!

**Review by RajkiranBagal — APPROVED** [review]

**Review by sandeshjangam — APPROVED** [review]

**Pull request merged**